File: 자료구조/Python/백준-큐-10845.py
from sys import stdin


def main():
    N = int(stdin.readline())

    queue = []

    for i in range(N):
        command = stdin.readline()
        if command.find("push") != -1:
            com = command.split()
            number = int(com[1])
            queue.append(number)
        elif command.find("front") != -1:
            if queue:
                print(queue[0])
            else:
                print(-1)
        elif command.find("back") != -1:
            if queue:
                print(queue[-1])
            else:
                print(-1)
        elif command.find("size") != -1:
            print(len(queue))
        elif command.find("pop") != -1:
            if queue:
                print(queue.pop(0))
            else:
                print(-1)
        elif command.find("empty") != -1:
            if queue:
                print(0)
            else:
                print(1)

if __name__ == "__main__":
    main()

File: 자료구조/C++/백준-큐-10845.cpp
# include <iostream>
# include <string>
# include <queue>
using namespace std;


int main()
{
    ios::sync_with_stdio(false);
    cin.tie(NULL);

    int N;
    cin >> N;

    string command;
    int number;
    queue<int> list;

    for (int i = 0; i < N; i++) 
    {
        cin >> command;
        if (command == "push") 
        {
            cin >> number;
            list.push(number);
        }
        else if (command == "front") 
        {
            if (!list.empty()) { cout << list.front() << endl; }
            else { cout << -1 << endl; }
        }
        else if (command == "back") 
        {
            if (!list.empty()) { cout << list.back() << endl; }
            else { cout << -1 << endl; }
        }
        else if (command == "size") 
        {
            cout << list.size() << endl;
        }
        else if (command == "empty") 
        {
            if (!list.empty()) { cout << 0 << endl; }
            else { cout << 1 << endl; }
        }
        else if (command == "pop") 
        {
            if (!list.empty()) { cout << list.front() << endl; list.pop(); }
            else { cout << -1 << endl; }
        }
    }
    
    return 0;
}